### PR TITLE
Faster grf

### PR DESCRIPTION
**Pull request by kmdeck**
Adds the model/equations for the equation `du = \theta*\nabla^2 u + \sigma dW` plus an EM timestepper.

Adds two scripts:
sim_analysis.jl: run a short sim & save a movie, histogram of a single pixel, and autocorrelation time plot
sim.jl: run a longer sim and save to HDF5

Options to save not at every step and account for spinup time. 
Correlated noise is supported.
Autocorrelation time ~100:, running for 1e6:

```
(julia1.9) katherinedeck@Katherines-iMac grf % julia --project sim.jl
155.063660 seconds (67.77 M allocations: 1.273 GiB, 0.09% gc time, 0.25% compilation time)
```

### CODE REVIEW

**Comment by kmdeck** [comment]
ac_1 0_0 5
single_pixel_1 0_0 5

**Comment by kmdeck** [comment]
I found that 
- at fixed \theta/\sigma, the autocorrelation time scales ~ inverse linearly with \theta.
- increasing \sigma at fixed \theta -> larger spread in steady state histogram
- varying sigma at fixed \theta does not change the results much for autocorrelation time. Maybe this is because (1) increasing sigma increases the variance in u, (2) the divergence of u therefore increases (linear in u?) , so (3) the magnitude of du/dt to du/dW is ~constant? this is just a hypothesis!


The first two observations make sense to start though

**Comment by kmdeck** [comment]
anim_1 0_0 5

File: Project.toml [hunk 1]
```diff
@@ -6,8 +6,11 @@ version = "0.1.0"
 [deps]
 DifferentialEquations = "0c46a032-eb83-5123-abaf-570d42b7fbaa"
 Distributed = "8ba89e20-285c-5b6f-9357-94700520ee1b"
+HDF5 = "f67ccb44-e63f-5c2f-98bd-6dc0ccc4ba2f"
+LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 Optim = "429524aa-4258-5aef-a3af-852621145aeb"
 OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
```
Comment (bischtob) [inline Project.toml:13]: Can we use Makie :) in the future :)?

Comment (bischtob) [reply Project.toml:13]: But maybe plots should be part of a separate example folder manifest. No need to do that now tho.

File: examples/grf/rhs.jl [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    W_corr::Vector{FT}
+end
+
+abstract type AbstractBoundaryConditions end
+
+struct Dirichlet{FT} <: AbstractBoundaryConditions
+    boundary_value::FT
+end
+
+function make_deterministic_tendency(model::LinearDiffusionSDE{FT, CM, Dirichlet{FT}}) where {FT, CM}
```
Comment (bischtob) [inline examples/grf/rhs.jl:20]: I can see some programming patterns here :).

**Review by bischtob — APPROVED** [review]
Looks very nice!

**Comment by bischtob** [comment]
I think you can squash an rebase and merge :).

File: Project.toml (hunk 1 above)
Comment (kmdeck) [reply Project.toml:13]: yes and yes!

**Pull request merged**